Lesson2.ipynb

```diff
@@ -648,6 +589,580 @@
+    "\n",
+    "print(\"Working days:\", workingDays)\n",
+    "print(\"And the original Tulpe still exist and hasn't been modified:\", weekDays)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Dictionaries\n",
+    "\n",
+    "List and Tuples are nice but become quickly limited when we want to manipulate more detailed and organised data.\n",
+    "\n",
+    "Let's write down some users pesonal data for example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "aliceDetails = [\n",
+    "    'Alice',    # name\n",
+    "    12,         # age\n",
+    "    ['reading', 'singing'] # hobbies\n",
+    "]\n",
+    "bobDetails = [\n",
+    "    24,\n",
+    "    ['dancing', 'drawing'],\n",
+    "    'Bob',\n",
+    "    '33 Mackenzie Road'\n",
+    "]\n",
+    "\n",
+    "print(\"Alice's age:\", aliceDetails[1])\n",
+    "print(\"Bob's age:\", bobDetails[0])\n",
+    "print(\"Bob's home address:\", bobDetails[3])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As a developer, it's easy to see what index tallies to the user's age.\n",
+    "\n",
+    "But, what if I have 100 users who have 100 different details each? And how do I retrieve Alice's home address? \n",
+    "\n",
+    "It would be a nightmare to retrieve the right detail accurately.\n",
+    "\n",
+    "That's why, for more detailed and organised data, we use a data structure called **Dictionaries**\n",
+    "\n",
+    "A Dictionary is a collection of key-value paired data. Unlike Lists, we use a label (called a key) as index, not a number.\n",
+    "\n",
+    "A Dictionnary is declared using curly brackets `{}` and a value is assigned to a key by using the colon `:` operator. E.g."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "alice_details = {\n",
+    "    'name': 'Alice', # key : value\n",
+    "    'age': 12,\n",
+    "    'hobbies': ['reading', 'singing']\n",
+    "}\n",
+    "bob_details = {\n",
+    "    'age': 24,\n",
+    "    'hobbies': ['dancing', 'drawing'],\n",
+    "    'name': 'Bob',\n",
+    "    'address': '33 Mackenzie Road'\n",
+    "}\n",
+    "\n",
+    "# dictionary[key] - We can access the value through the key, not the index\n",
+    "print(\"Alice's name:\", alice_details['name']) \n",
+    "print(\"Bob's name:\", bob_details['name']) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "No matter how the data is ordered, by using a key, we are sure to find the right data.\n",
+    "\n",
+    "Just like Lists, Dictionaries' values can be of any types. Even Lists and Dictionaries, which will allow us to create nested Dictionaries with even more detailed date.\n",
```

Comment:
"Even Lists and Dictionaries, which will allow us to create nested Dictionaries with even more detailed date." may read better as 
"Combining Lists and Dictionaries will allow us to create nested Dictionaries with even more detailed data."